docs/adr/0005-rapid-releases-and-moving-towards-a-monorepo.md

```diff
@@ -0,0 +1,83 @@
+* We want to move towards the future monorepo world
+* We are willing to accept the trade-offs involved with more frequent merges to application repositories.
+
+## Considered Options
+
+* **(A) Keep our current release process**
+    * Keep everything the same for a while and wait for other code to move into the monorepo.
+* **(B) Move to a rapid release process where changes in app-services are frequently merged into consumer applications**
+    * Update our branches to match how moz-central: we would have a `main`, `beta` and `release` branch, with nightly being built off of main, as well as `esr-*` branches for older ESR releases.
+    * Change it so our code lives in the repositories for consumer applications including firefox-android and firefox-ios.  We could either use a git submodule or vendor a copy of our code in to accomplish this.  Note: This is already planned for firefox-ios.
+    * Automate a nightly merge of our code our `main` branch to the `main` branch of the consumer app repos:
+       * Automate an application-services commit that updates our version number to something like `[firefox-version].nightly-[YYYYMMDD]`
+       * For each consumer application:
+         * Create a PR that updates the submodule commit id or vendors the new code in
+         * Set up an auto-merge system where if the test suite passes we automatically merge the PR.  As a fallback we could manually do this every day.
+         * Set up an alert that gets sent to `sync-bots` if the test suite fails.  This means we would need to identify why the tests failed and assign a developer to resolve the issue.  The expectation would be that the developer fixes things in time for the next nightly merge.
+    * In order to keep things running smoothly, if a breaking change requires serious review from the consumer teams, the PRs for that change should be approved and ready to merge before that breaking change is merged into application-services.  If we accidentally merge the change into application-services before the PRs are ready, then we should back the change out.
+    * Manually merge our code whenever we make changes to a `beta`, `release`, or `esr-*` branch.
+       * Commit a version bump to `[firefox-version].[N+1]`
+       * Make a PR in the corresponding branch for each application.  Create a tool to automate this.
+       * Collaborate with the mobile teams to get the PR merged.
+       * Merge the changes forward into the other branches.  If the change was committed to `release`, then we merge it into `beta`, and `main`.
+    * Continue to maintain our CHANGELOG, but with new versioning.  In main, we keep a rolling log of updates with the header `[firefox-version].0`
+    * On the monthly Firefox merge date, we do roll each branch forward:
+      - Create a new `esr-[version]` branch from `release`, if needed
+      - Merge `beta` into `release` (this should be a fast-forward).
+      - Bump the version number in main to `[firefox-version].0`
```

Comment:
Should version numbers have 2 components or 3?  I notice android-components just released `108.0.8` should we also have a middle component?  When would that not be 0?